src/app/pages/FastBtcPage/components/Withdraw/AddressForm.tsx

```diff
@@ -46,22 +48,37 @@ export const AddressForm: React.FC = () => {
     [set, value],
   );
 
-  const validate = useCallback(async (adr: string) => {
+  const validateAddr = useCallback(async (adr: string) => {
+    let validationReport,
+      validAddressCheck = false,
+      result;
     setAddressValidationState(AddressValidationState.LOADING);
-    const result = await contractReader.call(
-      'fastBtcBridge',
-      'isValidBtcAddress',
-      [adr],
-    );
+    const checkAddressValidity = validate(adr);
+    result = await contractReader.call('fastBtcBridge', 'isValidBtcAddress', [
```

Comment:
```suggestion
    const result = await contractReader.call('fastBtcBridge', 'isValidBtcAddress', [
```

Reply:
Can you tell me what's the difference between this check and check on line `56` (`validate()`)? Because it seems to me they should be the same thing.

Reply:
From FastBTC devs:

> it’s not enough to validate the prefix (starting characters) of the address, which is what the address validation smart contract does. Instead you must use a library that can decode the address to a script and check its checksum. After validating with the package, it would be good to also call the isValidBtcAddress function on the FastBTCBridge contract, to double-check.